src/AppKit/NSApplication.cs

```diff
@@ -70,10 +70,23 @@ public static void Init ()
 			// But can affect child xbuild processes, so unset
 			Environment.SetEnvironmentVariable ("MONO_CFG_DIR", "");
 
+			// custom initialization might have happened before native NSApplication code was full ready to be queried
+			// as such it's possible that `class_ptr` might be empty and that will make things fails later
+			// reference: https://github.com/xamarin/xamarin-macios/issues/7932
+			if (class_ptr == IntPtr.Zero)
+				ResetHandle ();
+
 			// TODO:
 			//   Install hook to register dynamically loaded assemblies
 		}
 
+		// separate method so it can be invoked without `Init` (if needed)
+		static void ResetHandle ()
```

Comment:
This looks like it should be called from 2 places? Who else sets class_ptr?

Reply:
not _should_ but _could_

You can't call `Init` twice so if, for a different custom initialization (hard to predict what people are/will be doing), you end up in the same situation (not initialized) then a workaround would be to call this `ResetHandle` method (thru reflection) at a later time.

Reply:
We could also have the linker eliminate it... since it should never be needed in scenarios where linking is supported.